subprojects/platform-native/src/main/java/org/gradle/nativeplatform/toolchain/internal/swift/SwiftcToolChain.java

```diff
@@ -91,8 +92,13 @@ private PlatformToolProvider createPlatformToolProvider(NativePlatformInternal t
         if (!result.isAvailable()) {
             return new UnavailablePlatformToolProvider(targetPlatform.getOperatingSystem(), result);
         }
+        SwiftcVersionResult swiftcMetaData = compilerMetaDataProvider.getSwiftcMetaData(compiler.getTool());
+        result.mustBeAvailable(swiftcMetaData);
+        if (!result.isAvailable()) {
+            return new UnavailablePlatformToolProvider(targetPlatform.getOperatingSystem(), result);
```

Comment:
We should probably have an integration test similar to `GccToolChainDiscoveryIntegrationTest` or `VisualCppToolChainDiscoveryIntegrationTest` that checks some of these failure cases for swift.  For instance, you could put a "swiftc" script on the tool search path that returns garbage when checking the version and validate that we get a reasonable error.

Reply:
Could you help me with that test? Or do you have an example where we already do this?

Reply:
I pushed a test that checks that we get good errors when 1) we don't find a swift compiler anywhere or 2) we find something but it doesn't return reasonable output.